src/ui/mainwindow.cpp

```diff
@@ -3096,3 +3117,15 @@ void MainWindow::keyPressEvent(QKeyEvent* event) {
     QMainWindow::keyPressEvent(event);
   }
 }
+
+// Change the state of action_next_album based when the repeat mode is
+// changed.
+void MainWindow::SetNextAlbumEnabled(PlaylistSequence::RepeatMode mode) {
+  if (mode == PlaylistSequence::Repeat_Track) {
+    if (ui_->action_next_album->isEnabled())
```

Comment:
Don't think you need to do this check.

Reply:
I put that in so the change to the action is only applied if action_next_album is enabled... So when the same selection is made again the action is not disabled again when it's already disabled.